src/main/java/picard/sam/markduplicates/util/ReadEndsForMarkDuplicates.java

```diff
@@ -37,7 +37,7 @@
        - int: read1ReferenceIndex, read1Coordinate, read2ReferenceIndex, read2Coordinate, duplicateSetSize
        - long: read1IndexInFile, read2IndexInFile
      */
-    protected static final int SIZE_OF = (1 * 1) + (5 * 2) + (5 * 4) + (8 * 2) + 1
+    protected static final int SIZE_OF = (1 * 1) + (5 * 2) + (5 * 4) + (8 * 2) + 1 + 1
             + 8 + // last 8 == reference overhead
             13; // This is determined experimentally with JProfiler
```

Comment:
That size doesn't look aligned (before or now), but maybe that doesn't matter.
Shouldn't (5 * 2) be (6 * 2) according to the comment?
Shouldn't (8 * 2) be (2 * 8) for consistency?
And why the +1s?
Maybe just determine the whole thing with the profiler.

Reply:
I have profiled the class with `java.lang.instrument.Instrumentation`. It shows that an implementation-specific (Java HotSpot(TM) 64-Bit Server) approximation of the amount of storage consumed by the `ReadEndsForMarkDuplicates` object equals to __72 bytes__.
Adding new boolean field has no influence on the approximated value.
Current value of `SIZE_OF` is __70__. So they are pretty equal.

After recounting this could looks like this:
```java
    // What do we need to store you ask?  Well, we need to store:
    protected static final int SIZE_OF =
              (2 * 1)   // - byte: orientation, orientationForOpticalDuplicates
            + (6 * 2)   // - short: libraryId, readGroup, tile, x, y, score
            + (5 * 4)   // - int: read1ReferenceIndex, read1Coordinate, read2ReferenceIndex, read2Coordinate, duplicateSetSize
            + (2 * 8)   // - long: read1IndexInFile, read2IndexInFile
            + 1         // - boolean: firstOfPair
            + 8         // last 8 == reference overhead
            + 13;       // This is determined experimentally with JProfiler
```
Which is equal to __72__. 
Is this OK, or should we just paste 72 from profiling?

Reply:
That makes sense to me.  Thanks.

Reply:
You can do that or summarize how to tell when SIZE_OF goes wrong and how to fix it,
but either way, thank you for straightening that out for us.